go-controller/pkg/ovn/address_set_test.go

```diff
@@ -0,0 +1,343 @@
+				}
+
+				var namespacesRes string
+				for _, n := range namespaces {
+					namespacesRes += fmt.Sprintf("name=%s\n", n.makeName())
+				}
+				fexec.AddFakeCmd(&ovntest.ExpectedCmd{
+					Cmd:    "ovn-nbctl --timeout=15 --data=bare --no-heading --columns=external_ids find address_set",
+					Output: namespacesRes,
+				})
+
+				err = forEachAddressSetUnhashedName(func(addrSetName, namespaceName, nameSuffix string) {
+					found := false
+					for _, n := range namespaces {
+						if addrSetName == n.makeName() {
+							found = true
+							Expect(namespaceName).To(Equal(n.namespace))
+							if n.suffix1 != "" {
+								Expect(nameSuffix).To(Equal(n.suffix1))
+							} else {
+								Expect(nameSuffix).To(Equal(""))
+							}
+						}
+					}
+					Expect(found).To(BeTrue())
+				})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fexec.CalledMatchesExpected()).To(BeTrue(), fexec.ErrorDesc)
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when creating an address set object", func() {
+		It("re-uses an existing address set and replaces IPs", func() {
+			app.Action = func(ctx *cli.Context) error {
+				const (
+					addr1 string = "1.2.3.4"
+					addr2 string = "5.6.7.8"
+				)
+
+				_, err := config.InitConfig(ctx, fexec, nil)
+				Expect(err).NotTo(HaveOccurred())
+
+				fexec.AddFakeCmd(&ovntest.ExpectedCmd{
+					Cmd:    "ovn-nbctl --timeout=15 --data=bare --no-heading --columns=_uuid find address_set name=a9625390261332436968",
+					Output: fakeUUID,
+				})
+				fexec.AddFakeCmdsNoOutputNoError([]string{
+					`ovn-nbctl --timeout=15 set address_set ` + fakeUUID + ` addresses="` + addr1 + `" "` + addr2 + `"`,
+				})
+
+				_, err = NewAddressSet("foobar", []net.IP{net.ParseIP(addr1), net.ParseIP(addr2)})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fexec.CalledMatchesExpected()).To(BeTrue(), fexec.ErrorDesc)
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("clears an existing address set of IPs", func() {
+			app.Action = func(ctx *cli.Context) error {
+				_, err := config.InitConfig(ctx, fexec, nil)
+				Expect(err).NotTo(HaveOccurred())
+
+				fexec.AddFakeCmd(&ovntest.ExpectedCmd{
+					Cmd:    "ovn-nbctl --timeout=15 --data=bare --no-heading --columns=_uuid find address_set name=a9625390261332436968",
+					Output: fakeUUID,
+				})
+				fexec.AddFakeCmdsNoOutputNoError([]string{
+					"ovn-nbctl --timeout=15 clear address_set " + fakeUUID + " addresses",
+				})
+
+				_, err = NewAddressSet("foobar", nil)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fexec.CalledMatchesExpected()).To(BeTrue(), fexec.ErrorDesc)
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("creates a new address set and sets IPs", func() {
+			app.Action = func(ctx *cli.Context) error {
+				const (
+					addr1 string = "1.2.3.4"
```

Comment:
adding an IPv6 test wouldn't make sure that the quotes stuff is right though, because we quote all addresses, not just IPv6 addresses, and this test isn't actually running against real OVN binaries, so it wouldn't actually fail if we didn't quote it

Reply:
it's probably still a good idea to add v6 tests, just not for the verification of the "quoting" behavior.

Reply:
but, i think that can be added later by someone else in one of the v6 PRs.

Reply:
let's table this and add @vjayaramrh make a note to ensure when we do the ipv6 unit tests across the board (with the new framework) that we add them for address sets as well